### PR TITLE
Add support for GHC 9.10

### PR DESCRIPTION
**Pull request by bgamari**
This requires https://github.com/haskell-hint/hint/pull/176.

### CODE REVIEW

**Review by martijnbastiaan — APPROVED** [review]

File: cabal.project [hunk 1]
```diff
@@ -1 +1,23 @@
...
+  type: git
+  location: https://github.com/clash-lang/clash-compiler
+  tag: 15dc344dfa091de14c63759c0b6ea107ca0fa892
+  subdir: clash-prelude
+
+source-repository-package
+  type: git
+  location: https://github.com/clash-lang/clash-compiler
+  tag: 15dc344dfa091de14c63759c0b6ea107ca0fa892
+  subdir: clash-prelude-hedgehog
```
Comment (martijnbastiaan) [inline cabal.project:23]:
```suggestion
-----------------------------------------------------------
-- clash-compiler for GHC 9.10
-----------------------------------------------------------
source-repository-package
  type: git
  location: https://github.com/clash-lang/clash-compiler
  tag: 15dc344dfa091de14c63759c0b6ea107ca0fa892
  subdir: clash-prelude
```

Comment (martijnbastiaan) [reply cabal.project:23]: Only `clash-prelude` is a dependency AFAIK. Was I wrong in my previous comment?

Comment (bgamari) [reply cabal.project]: Sigh, apologies; I had overlooked that bit of your comment. Fixed.

**Comment by bgamari** [comment]
The problem here appears to be that `clash` itself is incompatible with GHC 8.6. It's not entirely clear what to do about this. Perhaps we can drop support for this version in `circuit-notation`?

**Comment by martijnbastiaan** [comment]
@cchalmers Would you mind adding @bgamari as a collaborator?

**Comment by cchalmers** [comment]
I've invited bgaramri as a collaborator, glad to have you :smile:.

> The problem here appears to be that clash itself is incompatible with GHC 8.6. It's not entirely clear what to do about this. Perhaps we can drop support for this version in circuit-notation?

Unfortunately we're still stuck at ghc-8.6, trapped in a nix/ghc hole and our various attempts at updating have been stumped by weird bugs where the design stops working on the board.

Is this a recent change in clash? Can we get CI to use an old clash for ghc-8.6? I get that this could be annoying for development.

**Pull request merged**